### ISSUE TITLE
MagicNumber lint fix

### ISSUE THREAD

**Issue by nmgalo** [detekt/detekt#5758]
## Expected Behavior

I'm using Jetpack compose. I set `ignorePropertyDeclaration` to true, It works perfectly, But it has problem with if statement.
It would be better to ignore `ignorePropertyDeclaration` for if statements.

## Steps to Reproduce

```kotlin
Image(
    ...
    alpha = if (isSelected) 1f else 0.6f
)
```

`alpha = if (isSelected) 1f else 0.6f`: That's an error

`This expression contains a magic number. Consider defining it to a well named constant. [MagicNumber]` 

## Your Environment

* Version of detekt used: 1.22.0
* Version of Gradle used: 7.4.0

**Comment by cortinico** [comment]
That's a valid report,

> It would be better to ignore `ignorePropertyDeclaration` for if statements.

The snippet you reported though is about a named argument and not a property, so you'll need to use `ignoreNamedArgument`. The fix for it is #5837
